### PR TITLE
Replace haddock3-analysis-components with haddock3-ui package

### PR DESCRIPTION
**Pull request by sverhoeven**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md).

## Checklist

- [ ] Tests added for the new code
- [ ] Documentation added for the code changes
- [x] Does not break licensing
- [x] Does not add any dependencies, if it does please add a thorough explanation

## Summary of the Pull Request  

I made 2 npm packages 
1. https://www.npmjs.com/package/@i-vresse/haddock3-ui
2. https://www.npmjs.com/package/@i-vresse/haddock3-analysis-components

The `haddock3-ui` package now can do everything the `haddock3-analysis-components` can do. So I want to archive the `haddock3-analysis-components` GitHub repo.

This PR uses the haddock3-ui package for rendering tables of clusters and structures.

This PR also implements offline mode for the table in the report.html, no urls outside the local web server are requested.

## Related Issue
https://github.com/i-VRESSE/haddock3-ui/issues/6

## Additional Info

0 0 0 0_8000_analysis_10_caprieval_analysis_report html
0 0 0 0_8000_analysis_10_caprieval_analysis_report html (1)
0 0 0 0_8000_analysis_02_caprieval_analysis_report html

### CODE REVIEW

**Review by rvhonorato — DISMISSED** [review]

**Comment by VGPReys** [comment]
To which extend we could also ship the haddock3-ui when installing haddock3, so the `offline = true` would also be functional ?

File: src/haddock/libs/libplots.py [hunk 1]
```diff
@@ -980,7 +980,7 @@ def _css_styles_for_report():
     }
 
     """
-    css_link = "https://esm.sh/@i-vresse/haddock3-analysis-components@~0.4.1/dist/style.css"  # noqa:E501
+    css_link = "https://esm.sh/@i-vresse/haddock3-ui@~0.2.2/dist/index.css"
     table_css = f' <link href={css_link} rel="stylesheet" />'
```
Comment (VGPReys) [inline src/haddock/libs/libplots.py:984]:
```suggestion
    table_css = f' <link href="{css_link}" rel="stylesheet" />'
```

File: src/haddock/libs/libplots.py [hunk 2]
```diff
@@ -980,7 +980,7 @@ def _css_styles_for_report():
     }
```
Comment (VGPReys) [inline src/haddock/libs/libplots.py:980]: displace a little bit on the left the whole page
```suggestion
    }
    body {
      margin-left: 1em;
    }
```

**Comment by sverhoeven** [comment]
> To which extend we could also ship the haddock3-ui when installing haddock3, so the `offline = true` would also be functional ?

Working on offline mode in https://github.com/i-VRESSE/haddock3-ui/pull/9 .
How would we like to use the haddock3-ui js bundle? 
1. As an external file, 
    1. next to report.html, 
    2. somewhere within run_dir for reusing same js file.
2. JS embedded inside report.html

**Comment by VGPReys** [comment]
> > To which extend we could also ship the haddock3-ui when installing haddock3, so the `offline = true` would also be functional ?
> 
> Working on offline mode in [i-VRESSE/haddock3-ui#9](https://github.com/i-VRESSE/haddock3-ui/pull/9) . How would we like to use the haddock3-ui js bundle?
> 
>     1. As an external file,
>        
>        1. next to report.html,
>        2. somewhere within run_dir for reusing same js file.
> 
>     2. JS embedded inside report.html

Good question! design choice I guess.

- Can't it be somewhere in the `haddock3` installation repository ?  Like `haddock3/js-bundle/haddock3-ui.js`?

- Or maybe we want to have it contained in the `run_dir` somewhere, so results are self sufficient and can be separated from the code. I guess this is the best option. Could be placed in the `data` directory: `run_dir/data/ui/haddock3-ui.js`.

@rvhonorato , @amjjbonvin please provide input for this

**Comment by amjjbonvin** [comment]
How large is the js file?

**Comment by sverhoeven** [comment]
The js file is 1.5Mb and the css file 11Kb. 

@VGPReys indeed I was thinking of making it part of the haddock3 installation (using a similar mechanism as used to download cns executables in #1013) and then during analysis copy the css+js files to the run_dir, as the web server must have the css+js files in the directory it is hosting.

**Comment by sverhoeven** [comment]
<details>
<summary>Offline example</summary>

In examples/scoring dir created capri-scoring-test.offline.cfg with

```toml
# ===================================================================================
# CAPRI Scoring example
# ===================================================================================
# The Critical Assessment of PRedicted Interactions (CAPRI) experiment
#  aims to do test methods that model macromolecular interactions in
#  blind predictions based on the three-dimensional structures of proteins.
# For more information, please visit: https://www.ebi.ac.uk/pdbe/complex-pred/capri/
# ===================================================================================
run_dir = "capriscoring-test"

# execution mode
ncores = 40
mode = "local"
offline = true

# molecules to be scored (an ensemble PBD)
molecules = "data/T161-rescoring-ens.pdb"

# ===================================================================================
[topoaa]

[emscoring]

[caprieval]

[clustfcc]
min_population = 2

[seletopclusts]
top_cluster = 1
top_models = 2

[mdscoring]
per_interface_scoring = true

[clustfcc]
min_population = 2

[seletopclusts]

[caprieval]

# ===================================================================================

```

Ran with

```shell
rm -r capriscoring-test
haddock3 capri-scoring-test.offline.cfg
mkdir capriscoring-test/data/ui
curl -o capriscoring-test/data/ui/report.bundle.js https://cdn.jsdelivr.net/npm/@i-vresse/haddock3-ui@~0.3.0/dist/report.bundle.js
curl -o capriscoring-test/data/ui/index.css https://cdn.jsdelivr.net/npm/@i-vresse/haddock3-ui@~0.3.0/dist/index.css
```

</details>

File: src/haddock/libs/libplots.py (hunk 1 above)
Comment (sverhoeven) [reply src/haddock/libs/libplots.py]: oeps

**Comment by sverhoeven** [comment]
@rvhonorato At https://github.com/haddocking/haddock3/pull/1025#issuecomment-2363320459 I am working on rendering tables in offline mode. To do that I need to copy a css and js file from the haddock3 installation. 

How can we make those files part of the haddock3 installation? I see several options:

1. Use custombuild step similar to https://github.com/haddocking/haddock3/pull/1013 to download files.
2. have the files be part of the repo and have a script to update them.
3. Have a haddock3-ui python package that contains the js and css files and is a dependency of haddock3 package.

What has your preference?

**Comment by sverhoeven** [comment]
I went for option 2 so a 1.5Mb js file will be part of the repo.

**Review by mgiulini — DISMISSED** [review]
both unclustered and clustered tables look good!

**Comment by sverhoeven** [comment]
> I see that `plotly` menu is on top of the cluster legend (see screenshot) Screenshot 2024-09-18 at 09-08-41 Analysis report of step 9_caprieval

Fixed with https://github.com/i-VRESSE/haddock3/commit/d24a510d70727488465ce28f01a5bef3a649b92c

**Review by mgiulini — DISMISSED** [review]

**Comment by sverhoeven** [comment]
@VGPReys you beat me with #1041,  so I will solve the conflicts.

**Comment by VGPReys** [comment]
> @VGPReys you beat me with #1041, so I will solve the conflicts.

Oops...
I did not realized that you also worked on it !

**Comment by sverhoeven** [comment]
> > @VGPReys you beat me with #1041, so I will solve the conflicts.
> 
> Oops... I did not realized that you also worked on it !

No worries, conflicts have been resolved.

**Review by mgiulini — APPROVED** [review]

**Pull request merged**